packages/cli/src/environments/variables/variables.service.ts

```diff
@@ -1,20 +1,39 @@
 import type { Variables } from '@db/entities/Variables';
 import { collections } from '@/Db';
+import { CacheService } from '@/services/cache.service';
+import { Service } from 'typedi';
 
+@Service()
 export class VariablesService {
-	static async getAll(): Promise<Variables[]> {
-		return collections.Variables.find();
+	constructor(private cacheService: CacheService) {}
+
+	async getAll(): Promise<Variables[]> {
+		const variables = await this.cacheService.get('variables', {
+			async refreshFunction() {
+				// TODO: log refresh cache metric
+				return collections.Variables.find();
+			},
+		});
+		return variables as Variables[];
 	}
 
-	static async getCount(): Promise<number> {
-		return collections.Variables.count();
+	async getCount(): Promise<number> {
+		return (await this.getAll()).length;
 	}
 
-	static async get(id: string): Promise<Variables | null> {
-		return collections.Variables.findOne({ where: { id } });
+	async get(id: string): Promise<Variables | null> {
+		const variables = await this.getAll();
+		return variables.find((variable) => variable.id === id) ?? null;
 	}
 
-	static async delete(id: string): Promise<void> {
+	async delete(id: string): Promise<void> {
 		await collections.Variables.delete(id);
+		await this.updateCache();
```

Comment:
```suggestion
		void this.updateCache();
		return this.variablesRepository.delete(id);
```

Reply:
updating the cache first and then deleting would leave the deleted item in the cache...

Reply:
You're right, confused this one!